src/yunohost/settings.py

```diff
@@ -0,0 +1,236 @@
+def settings_list():
+    """
+    List all entries of the settings
+
+    """
+    return _get_settings()
+
+
+def settings_set(key, value):
+    """
+    Set an entry value in the settings
+
+    Keyword argument:
+        key -- Settings key
+        value -- New value
+
+    """
+    settings = _get_settings()
+
+    if key not in settings:
+        raise MoulinetteError(errno.EINVAL, m18n.n(
+            'global_settings_key_doesnt_exists', settings_key=key))
+
+    key_type = settings[key]["type"]
+
+    if key_type == "bool":
+        if not isinstance(value, bool):
+            raise MoulinetteError(errno.EINVAL, m18n.n(
+                'global_settings_bad_type_for_setting', setting=key,
+                received_type=type(value).__name__, expected_type=key_type))
+    elif key_type == "int":
+        if not isinstance(value, int) or isinstance(value, bool):
+            raise MoulinetteError(errno.EINVAL, m18n.n(
+                'global_settings_bad_type_for_setting', setting=key,
+                received_type=type(value).__name__, expected_type=key_type))
+    elif key_type == "string":
+        if not isinstance(value, basestring):
+            raise MoulinetteError(errno.EINVAL, m18n.n(
+                'global_settings_bad_type_for_setting', setting=key,
+                received_type=type(value).__name__, expected_type=key_type))
+    elif key_type == "enum":
+        if value not in settings[key]["choices"]:
+            raise MoulinetteError(errno.EINVAL, m18n.n(
+                'global_settings_bad_choice_for_enum', setting=key,
+                received_type=type(value).__name__,
+                expected_type=", ".join(settings[key]["choices"])))
+    else:
+        raise MoulinetteError(errno.EINVAL, m18n.n(
+            'global_settings_unknown_type', setting=key,
+            unknown_type=key_type))
+
+    settings[key]["value"] = value
+
+    _save_settings(settings)
+
+
+def settings_reset(key):
+    """
+    Set an entry value to its default one
+
+    Keyword argument:
+        key -- Settings key
+
+    """
+    settings = _get_settings()
+
+    if key not in settings:
+        raise MoulinetteError(errno.EINVAL, m18n.n(
+            'global_settings_key_doesnt_exists', settings_key=key))
+
+    settings[key]["value"] = settings[key]["default"]
+    _save_settings(settings)
+
+
+def settings_reset_all():
+    """
+    Reset all settings to their default value
+
+    Keyword argument:
+        yes -- Yes I'm sure I want to do that
+
+    """
+    settings = _get_settings()
+
+    # For now on, we backup the previous settings in case of but we don't have
+    # any mecanism to take advantage of those backups. It could be a nice
+    # addition but we'll see if this is a common need.
+    # Another solution would be to use etckeeper and integrate those
+    # modification inside of it and take advantage of its git history
+    old_settings_backup_path = SETTINGS_PATH_OTHER_LOCATION % datetime.now().strftime("%F_%X")
```

Comment:
it seems inconsistent to me that the backup is done only with reset-all. Either we support backup, and then we perform a backup at each user modification, either we don't support backup at all for now.

i don't have a strong opinion on whether to perform backup or not. as long as we don't have a huge set of settings, I tend to think we can do without.